### ISSUE TITLE
SimpleXMLElement - SimpleXMLIterator::var is not defined - After property_exists check

### ISSUE THREAD

**Issue by lsv** [vimeo/psalm#6686]
With the following, I get a 

Instance property SimpleXMLIterator::$projectFiles is not defined (see https://psalm.dev/039)

Now, I already check that `$projectFiles` is a property, with my `property_exists` method

```php
<?php
$xml = <<<XML
<?xml version="1.0"?>
<psalm>
    <projectFiles>
        <directory name="src" />
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
</psalm>
XML;

$xmlObject = simplexml_load_string($xml);
if ($xmlObject instanceof SimpleXMLElement) {
    $projectFiles = $xmlObject->children('projectFiles', true);
    if (property_exists($projectFiles, 'projectFiles')) {
        if ($projectFiles->projectFiles instanceof SimpleXMLElement) {
            return true;
        }
    }
}
```

Isnt this a false positive call, or is there a way to get rid of that check?

**Comment by orklah** [comment]
You can add 
```xml
<universalObjectCrates>
        <class name="SimpleXMLIterator"/>
    </universalObjectCrates>
```

in your config file to resolve this (Psalm will consider any property as existing).

It should probably be something more integrated in Psalm though so I'll let this issue open

**Comment by weirdan** [comment]
It should.
